proto/cosmos/app/v1alpha1/query.proto

```diff
@@ -3,12 +3,20 @@ syntax = "proto3";
 package cosmos.app.v1alpha1;
 
 import "cosmos/app/v1alpha1/config.proto";
+import "google/protobuf/descriptor.proto";
+import "cosmos/query/v1/query.proto";
 
 // Query is the app module query service.
 service Query {
 
   // Config returns the current app config.
   rpc Config(QueryConfigRequest) returns (QueryConfigResponse) {}
+
+  // FileDescriptorSet returns the full FileDescriptorSet of the app in order
+  // to enable easier generation of dynamic clients.
+  rpc FileDescriptorSet(QueryFileDescriptorSetRequest) returns (QueryFileDescriptorSetResponse) {
```

Comment:
cant this be done through grpc reflect?

Reply:
![Screenshot 2022-10-19 at 18 41 18](https://user-images.githubusercontent.com/24299864/196752750-24c2300f-8b82-42f8-a9ee-d64731877b81.png)


could we actually attach this to an existing service that has other descriptors?

Reply:
grpc reflection doesn't have an easy way to get _all_ the descriptors. The other reflection APIs are very alpha IMHO and the app query service is designed to be a bit cleaner. I guess it could go in another reflection query service though if needed

Reply:
Also I would like to mark this as a solid `v1` wherever it goes. 

Maybe we could have a query to get all file descriptor names and grpc reflection could do the rest although the current implementation is broken and I would need to reimplement all of it which is more work.

Reply:
In the interest of splitting this into smaller PRs, let's continue this in #13597